geoutils/raster/raster.py

```diff
@@ -379,15 +435,18 @@ def _check_cast_array_raster(
                 "same grid and CRS than raster2."
             )
 
-    else:
+    # If other is an array
+    elif isinstance(other, np.ndarray):
 
-        # The shape compatibility should be valid even when squeezing
-        if isinstance(input1, np.ndarray):
-            input1 = input1.squeeze()
-        elif isinstance(input2, np.ndarray):
-            input2 = input2.squeeze()
+        # Squeeze first axis of other data if possible
+        if len(other.shape) == 3 and other.shape[0] == 1:
```

Comment:
No big deal, but you could use `other.ndim`.

Reply:
You're right, I changed it in the entire package to be more consistent :wink: